internal/controller/utils.go

```diff
@@ -16,11 +20,21 @@ func updateDeploymentAnnotations(deployment *appsv1.Deployment, annotations map[
 	}
 }
 
-func hashBytes(sourceStr []byte) (string, error) {
-	hashFunc := sha256.New()
-	_, err := hashFunc.Write(sourceStr)
+func getSHAfromConfigmap(configmap *corev1.ConfigMap) string {
+	values := []string{}
+	for k, v := range configmap.Data {
+		values = append(values, k+"="+v)
+	}
+	sort.Strings(values)
+	return generateSHA(strings.Join(values, ";"))
+}
+
+func generateSHA(data string) string {
+	hasher := sha256.New()
+	_, err := io.WriteString(hasher, data)
 	if err != nil {
-		return "", fmt.Errorf("failed to generate hash %w", err)
+		return ""
```

Comment:
@vbelouso In case of an error , we want to reconcile again right ? we are just returning ""